test/integration/targets/file/tasks/main.yml

```diff
@@ -16,134 +16,229 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-- set_fact: output_file={{output_dir}}/foo.txt
+- set_fact:
+    output_file: '{{ output_dir }}/foo.txt'
 
 - name: prep with a basic copy
-  copy: src=foo.txt dest={{output_file}}
+  copy:
+    src: foo.txt
+    dest: '{{ output_file }}'
 
 - name: verify that we are checking a file and it is present
-  file: path={{output_file}} state=file
+  file:
+    path: '{{ output_file }}'
+    state: file
   register: file_result
 
 - name: verify that the file was marked as changed
   assert:
     that:
-      - "file_result.changed == false"
-      - "file_result.state == 'file'"
+      - file_result.changed == false
+      - file_result.state == 'file'
 
 - name: verify that we are checking an absent file
-  file: path={{output_dir}}/bar.txt state=absent
+  file:
+    path: '{{ output_dir }}/bar.txt'
+    state: absent
   register: file2_result
 
 - name: verify that the file was marked as changed
   assert:
     that:
-      - "file2_result.changed == false"
-      - "file2_result.state == 'absent'"
+      - file2_result.changed == false
+      - file2_result.state == 'absent'
 
 - name: verify we can touch a file
-  file: path={{output_dir}}/baz.txt state=touch
+  file:
+    path: '{{ output_dir }}/baz.txt'
+    state: touch
   register: file3_result
 
 - name: verify that the file was marked as changed
   assert:
     that:
-      - "file3_result.changed == true"
-      - "file3_result.state == 'file'"
-      - "file3_result.mode == '0644'"
+      - file3_result.changed == true
+      - file3_result.state == 'file'
+      - file3_result.mode == '0644'
 
 - name: change file mode
-  file: path={{output_dir}}/baz.txt mode=0600
+  file:
+    path: '{{ output_dir }}/baz.txt'
+    mode: 0600
   register: file4_result
 
 - name: verify that the file was marked as changed
   assert:
     that:
-      - "file4_result.changed == true"
-      - "file4_result.mode == '0600'"
+      - file4_result.changed == true
+      - file4_result.mode == '0600'
 
 - name: change ownership and group
-  file: path={{output_dir}}/baz.txt owner=1234 group=1234
+  file:
+    path: '{{ output_dir }}/baz.txt'
+    owner: 1234
+    group: 1234
+
+# NOTE:File attributes only work on Linux outside docker (AuFS)
+- name: test attributes when not inside docker
+  when: >
```

Comment:
@sivel I found os.chflags() in the code, but wasn't sure this would work on MacOSX and FreeBSD.
WIll change.